recipes/cairo/meson/conanfile.py

```diff
@@ -101,24 +93,24 @@ def requirements(self):
         if self.options.with_zlib:
             self.requires("zlib/[>=1.2.11 <2]")
         if self.options.with_freetype:
-            self.requires("freetype/2.13.0")
+            self.requires("freetype/2.13.0", transitive_headers=True)
         if self.options.with_fontconfig:
-            self.requires("fontconfig/2.14.2")
+            self.requires("fontconfig/2.14.2", transitive_headers=True)
         if self.options.with_png:
             self.requires("libpng/1.6.40")
         if self.options.with_glib:
             self.requires("glib/2.78.0")
-        if self.settings.os == "Linux":
+        if self.settings.os in ["Linux", "FreeBSD"]:
             if self.options.with_xlib or self.options.with_xlib_xrender or self.options.with_xcb:
-                self.requires("xorg/system")
+                self.requires("xorg/system", transitive_headers=True)
         if self.options.get_safe("with_opengl") == "desktop":
-            self.requires("opengl/system")
+            self.requires("opengl/system", transitive_headers=True)
             if self.settings.os == "Windows":
                 self.requires("glext/cci.20210420")
                 self.requires("wglext/cci.20200813")
                 self.requires("khrplatform/cci.20200529")
         if self.options.get_safe("with_opengl") and self.settings.os in ["Linux", "FreeBSD"]:
-            self.requires("egl/system")
+            self.requires("egl/system", transitive_headers=True)
```

Comment:
Can we instead require the `egl` from `libglvnd` here, or does it require Mesa too?

Reply:
I'm not that familiar with either of these libraries or Mesa. You will have to be more specific, sorry. :smile:

Reply:
I can apply that change though, if it's necessary, sure. Is `transitive_libs=True` not required, though?

Reply:
Looking at it again, it should probably stay as `egl/system`. That package just installs `libglvnd` and/or the Mesa EGL drivers, but I think it mostly depends on Mesa. I'm working on a Mesa package which will hopefully allow us to specify a dependency on a non-system Conan package. Hope I didn't confuse you too much.